tests/mocha/connection_checker_test.js

```diff
@@ -341,4 +341,117 @@ suite('Connection checker', function() {
       chai.assert.isFalse(this.checker.doTypeChecks(this.con1, this.con2));
     });
   });
+  suite('Dragging Checks', function() {
+    suite('Stacks', function() {
+      setup(function() {
+        this.workspace = Blockly.inject('blocklyDiv');
+        // Load in three blocks: A and B are connected (next/prev); B is unmovable.
+        Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(`<xml xmlns="https://developers.google.com/blockly/xml">
+        <block type="text_print" id="wlc#l2D-,6mSLKY=E$VL" x="-76" y="-112">
+          <next>
+            <block type="text_print" id="T9cdR}mwDn3Xl!*g|mw8" movable="false">
+              <comment pinned="true" h="32" w="88.5333251953125">Immovable</comment>
+            </block>
+          </next>
+        </block>
+        <block type="text_print" id="ZgFlirwB{~}%l/BZ+j?R" x="47" y="-118"/>
+      </xml>`), this.workspace);
+      [this.blockA, this.blockB, this.blockC] = this.workspace.getAllBlocks(true);
+      this.checker = this.workspace.connectionChecker;
+      });
+
+      test('Connect a stack', function() {
+        // block C is not connected to block A; both are movable.
+        chai.assert.isTrue(
+          this.checker.doDragChecks(
+            this.blockC.nextConnection, this.blockA.previousConnection, 9000),
+          'Should connect two compatible stack blocks');
+      });
+
+      test('Do not splice into unmovable stack', function() {
+        // Try to connect blockC above blockB. It shouldn't work because B is not movable
+        // and is already connected to A's nextConnection.
+        chai.assert.isFalse(
+          this.checker.doDragChecks(
+            this.blockC.previousConnection, this.blockA.nextConnection, 9000),
+          'Should not splice in a block above an unmovable block');
+      });
+
+      test('Connect to bottom of unmovable stack', function() {
+        // Try to connect blockC below blockB.
+        // This is allowed even though B is not movable because it is on B's nextConnection.
+        chai.assert.isTrue(
+          this.checker.doDragChecks(
+            this.blockC.previousConnection, this.blockB.nextConnection, 9000),
+          'Should connect below an unmovable stack block');
+      });
+
+      test('Connect to unconnected unmovable block', function() {
```

Comment:
Is this the desired behavior? Because then they get "glued" together right, and the user can't disconnect them again, or insert anything between them? Or is it possible to drag away the new top block? (blockC in this case).

Similar question for the row block situation.

Reply:
So, I'm not sure of the original "desired" behavior. I chose to just match the existing behavior of the row blocks. I am erring on the side of changing as little as possible though I don't see why we couldn't change it if someone asked us to in the future.

I expect this to matter very little as the chances someone prepopulates the workspace with an unmovable block with an empty previous connection seem small. The normal circumstances for doing this would be a "start block" which doesn't have a previous connection, so it doesn't matter, or blocks that are connected to such a start block, which is fixed by this PR.